src/codebook/util/indicateLoading.js

```diff
@@ -0,0 +1,15 @@
+import { select as d3select } from 'd3';
+
+export default function indicateLoading(codebook, element, callback) {
+  codebook.loadingIndicator.style('display', 'block');
+  //wait by the centisecond until the loading indicator is visible
+  const loading = setInterval(() => {
+    const laidOut = d3select(element).property('offsetwidth') > 0;
+    if (!laidOut) {
```

Comment:
Check that the target element does not have "display:none" before attempting to render?